tools/runk/src/main.rs

```diff
@@ -0,0 +1,231 @@
+                .about("create and run a container")
+                .arg(&id)
+                .arg(&bundle)
+                .arg(&pid_file)
+                .arg(&console_socket)
+                .arg(&no_pivot),
+        )
+        // create command
+        .subcommand(
+            SubCommand::with_name("create")
+                .about("create a container")
+                .arg(&id)
+                .arg(&bundle)
+                .arg(&pid_file)
+                .arg(&console_socket)
+                .arg(&no_pivot),
+        )
+        // start command
+        .subcommand(
+            SubCommand::with_name("start")
+                .about("start executes the user defined process in a created container")
+                .arg(&id),
+        )
+        // kill command
+        .subcommand(
+            SubCommand::with_name("kill")
+                .about("kill sends the sepcified signal (default; SIGTERM) to the container's init process")
+                .arg(&id)
+                .arg(Arg::with_name("signal")
+                     .default_value("SIGTERM")
+                     .required(true)
+                     .takes_value(true)
+                     .help("the signal to be sent to the init process"))
+                .arg(Arg::with_name("all")
+                     .long("all")
+                     .short("a")
+                     .help("send the specified signal to all processes inside the container")),
+        )
+        // delete command
+        .subcommand(
+            SubCommand::with_name("delete")
+                .about("delete any resources held by the container often used with detached container")
+                .arg(&id)
+                .arg( Arg::with_name("force")
+                      .long("force")
+                      .short("f")
+                      .help("forcibly deletes the container if it is still running (uses SIGKILL)")),
+        )
+        // state command
+        .subcommand(
+            SubCommand::with_name("state")
+                .about("output the state of a container")
+                .arg(&id)
+        )
+        .get_matches();
+
+    // Set log level
+    let mut log_level = DEFAULT_LOG_LEVEL;
+    if matches.is_present("debug") {
+        log_level = slog::Level::Debug;
+    }
+
+    // Set up log file
+    let logger;
+    let _logger_async_guard;
+    if let Some(log_file) = matches.value_of("log") {
+        let log_writer = OpenOptions::new()
+            .write(true)
+            .read(true)
+            .create(true)
+            .truncate(true)
+            .open(log_file)?;
+
+        let (logger_local, logger_async_guard_local) =
+            logging::create_logger(crate_name!(), crate_name!(), log_level, log_writer);
+
+        logger = logger_local;
+        _logger_async_guard = logger_async_guard_local;
+    } else {
+        logger = slog::Logger::root(slog::Discard, o!());
+    }
+
+    let state_root = Path::new(matches.value_of("root").unwrap());
```

Comment:
Yes, but the problem is you have two pieces of code which guarantee state, but they are disconnected (and thus likely to deviate over time). We're trying to remove as many `unwrap()` and `expect()` calls as practical and this can be avoided with a slight reworking of the code.

Reply:
I understand. I removed all `unwrap()` except for test.